src/api/goods.ts

```diff
@@ -3,17 +3,25 @@ import { Good } from '../types/Good';
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL)
-    .then(response => response.json());
+export enum QueryTypes {
+  firstFiveGoods = 'firstFiveGoods',
+  allRedGoods = 'allRedGoods',
+  allGoods = 'allGoods',
 }
 
-export const get5First = () => {
-  return getAll()
-    .then(goods => goods); // sort and get the first 5
-};
 
-export const getRedGoods = () => {
-  return getAll()
-    .then(goods => goods); // get only red
+export const getGoods = async (query: QueryTypes) => {
+  const result = await fetch(API_URL);
+  const goodsList: Good[] = await result.json();
```

Comment:
nope, it should be three separate requests
When user clicks on get5First you should make one request and get only first five good, when user clicks getRedGoods you should make one request and get only red ones, ...
Don't make one request where you are fetching all info

Reply:
https://user-images.githubusercontent.com/16674930/198024148-8edecf2d-160f-4c57-8528-ccabd0368464.mov

Reply:
Yes, you make one request, but it is the same request for each button. Each button must have a separate function that makes a request to the server